### PR TITLE
Core integration playground

### PR DESCRIPTION
**Pull request by ChenyuLInx**
This need to use https://github.com/dbt-labs/dbt-core/tree/chenyu/manifest-hacking branch of core to work

### CODE REVIEW

File: dbt_server/views.py [hunk 1]
```diff
@@ -389,6 +401,119 @@ async def run_operation_async(
...
+async def async_entry(
+    command: str,
+    request: Request,
+    background_tasks: BackgroundTasks,
+    # commons: list = Depends(common_parameters),
+    # args: list,
+    response_model=schemas.Task,
+    db: Session = Depends(crud.get_db),
+):
+    if command not in dbt.commands.keys():
```
Comment (ChenyuLInx) [inline dbt_server/views.py:450]: We don't really need the `.keys()` since it is a dictionary

File: dbt_server/views.py [hunk 2]
```diff
@@ -388,6 +404,44 @@ async def run_operation_async(
...
+    # response_model=schemas.Task,
+    # db: Session = Depends(crud.get_db),
+):  
+    # example request:
+    # http://127.0.0.1:8580/async/dbt/?args=--printer-width&args=100&args=run&args=--project-dir&args=/Users/chenyuli/git/python-models-test-project-small
+    
+    set_invocation_args(args)
+    # create invocation ctx, this will take care of validation of args
+    try:
+        ctx = DBTContext(args)
```
Comment (ChenyuLInx) [inline dbt_server/views.py:423]: In the future we plan to add more function to DBTContext, you can imagine we would be able to do
``` python
profile = new_core_api.load_profile(profile_json)

ctx.set_profile(profile)
ctx.set_project(project) # project and profile are being used to construct RuntimeConfig
ctx.set_manifest(manifest)
ctx.set_logger(custom_logger)
```

File: dbt_server/views.py [hunk 3]
```diff
@@ -388,6 +404,44 @@ async def run_operation_async(
...
+    # background_tasks: BackgroundTasks,
+    args: List[str]= Query(None),
+    # request: Request,
+    # commons: list = Depends(common_parameters),
+    # args: list,
+    # response_model=schemas.Task,
+    # db: Session = Depends(crud.get_db),
+):  
+    # example request:
+    # http://127.0.0.1:8580/async/dbt/?args=--printer-width&args=100&args=run&args=--project-dir&args=/Users/chenyuli/git/python-models-test-project-small
```
Comment (ChenyuLInx) [inline dbt_server/views.py:418]: put the args of body

File: dbt_server/views.py [hunk 4]
```diff
@@ -375,34 +257,44 @@ async def seed_async(
...
+    # with body {"state_id": "123", "command":["run"]}
+    state = StateController.load_state(args.state_id, args)
+
+    task_id = str(uuid.uuid4())
+    log_path = filesystem_service.get_path(state.state_id, task_id, "logs.stdout")
+
+    task = schemas.Task(
+        task_id=task_id,
+        state=TaskState.PENDING,
+        command="dbt run-operation",
```
Comment (racheldaniel) [inline dbt_server/views.py:287]:
```suggestion
        command=(" ").join(args.command),
```

**Review by jp-dbt — APPROVED** [review]

**Pull request merged**